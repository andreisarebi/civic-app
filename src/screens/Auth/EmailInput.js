import React from 'react';
import PropTypes from 'prop-types';
import CivicTextInput from '../../util/components/CivicTextInput';

const EmailInput = props => {
  return (
    <CivicTextInput
      autoCapitalize="none"
      autoCorrect={false}
      autoFocus={false}
      clearButtonMode="while-editing"
      keyboardType="email-address"
      onChangeText={props.onChangeText}
      value={props.value}
      placeholder={props.placeholder || 'E-mail address'}
    />
  );
};

EmailInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};

export default EmailInput;
