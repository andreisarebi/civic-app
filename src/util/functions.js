import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
