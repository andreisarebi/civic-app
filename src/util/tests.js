import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
