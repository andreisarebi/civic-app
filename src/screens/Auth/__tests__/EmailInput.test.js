import { createSnapshot } from '../../../util/tests';
import EmailInput from '../EmailInput';

describe('EmailInput', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(EmailInput, {
      value: 'civic@civic.com',
      onChangeText: jest.fn(),
    });
    expect(snapshot).toMatchSnapshot();
  });
  it('renders placeholder text when there is no value', () => {
    const snapshot = createSnapshot(EmailInput, { value: '', onChangeText: jest.fn() });
    expect(snapshot).toMatchSnapshot();
  });
});
