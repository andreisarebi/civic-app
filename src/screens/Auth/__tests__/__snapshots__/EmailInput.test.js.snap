// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput renders correctly 1`] = `
<CivicTextInput
  autoCapitalize="none"
  autoCorrect={false}
  autoFocus={true}
  clearButtonMode="while-editing"
  keyboardType="email-address"
  onBlur={undefined}
  onChangeText={[MockFunction]}
  onFocus={undefined}
  placeholder="E-mail address"
  value="civic@civic.com"
/>
`;

exports[`EmailInput renders placeholder text when there is no value 1`] = `
<CivicTextInput
  autoCapitalize="none"
  autoCorrect={false}
  autoFocus={true}
  clearButtonMode="while-editing"
  keyboardType="email-address"
  onBlur={undefined}
  onChangeText={[MockFunction]}
  onFocus={undefined}
  placeholder="E-mail address"
  value=""
/>
`;
